examples/file_browser.html

```diff
@@ -0,0 +1,119 @@
+
+-->
+
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <script src="../umd/regular-table.js"></script>
+    <link rel='stylesheet' href="../css/material.css">
+</head>
+
+<body>
+
+    <regular-table></regular-table>
+
+    <script>
+
+        const NUM_ROWS = 100;
+        const NUM_DIR = 10;
+        const NUM_FILE = NUM_ROWS - NUM_DIR;
+
+        const COLUMN_NAMES = ['modified', 'writable'];
+
+        const formatter = new Intl.DateTimeFormat("en-us");
+
+        const _contents_cache = new Map();
+        function getContents(path, expand) {
+            // infinite recursive mock contents
+            let contents;
+            if (_contents_cache.has(path)) {
+                contents = _contents_cache.get(path);
+            } else {
+                contents = {path, modified: new Date(1000), writable: false};
+                _contents_cache.set(path, contents);
+            }
+
+            if (!expand || 'contents' in contents) {
+                return contents;
+            }
+
+            let sub = []
+            for (let i = 0; i < NUM_ROWS; i ++) {
+                sub.push({
+                    path: [...path, i < NUM_DIR ? `${i}/` : `${i}.txt`],
+                    modified: new Date(1000*86400*i),
+                    writable: false,
+                });
+            }
+
+            contents.contents = sub;
+            return contents;
+        }
+
+        function insertContents(rix, arr) {
+            const contents = getContents(arr[rix].path, true);
+            arr.splice(rix, 0, ...contents.contents);
+        }
+
+        function popContents(rix, arr) {
+            const {path} = arr[rix]
+            const contents = getContents(path, true);
+
+            // pop out the contents of the collapsed node and any expanded subnodes
+            let npop = contents.contents.length;
+            let check_ix = rix + 1 + npop;
+            while (arr[check_ix++].path.length > path.length) {
+                npop++;
+            }
+
+            arr.splice(rix + 1, npop);
+        }
+
+        let rows = getContents([], true).contents;
+
+        async function file_browser_model(start_col, start_row, end_col, end_row) {
+            const data = [];
+            for (let cix = start_col; cix < end_col; cix ++) {
+                const name = COLUMN_NAMES[cix];
+                data.push(rows.slice(start_row, end_row).map(c => {
+                    return name === 'modified' ? formatter.format(c[name]) : c[name];
+                }));
+            }
+
+            return {
+                num_rows: rows.length,
+                num_columns: COLUMN_NAMES.length,
+                column_indices: COLUMN_NAMES.map(col => [col]),
+                row_indices: rows.slice(start_row, end_row).map(c => c['path']),
+                data,
+            };
+        }
+
+        // for tree support
```

Comment:
`collapse` and `expand` are perspective API holdovers that will be removed.  
* For the time being, they should be refactored to `__collapse` and `__expand` respectively, and please also apply this fix to `perspective.html` to map these back to `perspective.View()`.
* Eventually, use `.addEventListener("click", =>` on the `regular-table` and check for button contact in the example.
* In the far future, remove the +/- rendering entirely and implement this as a custom cell with templating.

Also, can we simplify this by replacing inlining and removing the ambiguously-named `popContents()` and `insertContents()`?

Reply:
Done! I added `__collapse`, `__expand`, and `__set_depth` to `perspective.html`, and added corresponding code in `RegularVirtualTableViewModel.draw` in `scroll_panel.js`.

I ran into an issue with the perspective example where all of the row indices were rendering as `NaN`. This turned out to be because all of the columns types listed in `__schema` were `integer`, and the tree row header tries to format the row header text based on that type:

https://github.com/jpmorganchase/regular-table/blob/50a7ea908d91cb969961b2bdd00262ef5c8f3229/src/js/tree_row_header.js#L43

All of this formatting stuff is going to get thrown out or significantly refactored, right? For right now, I was able to "fix" this by setting `__schema` to `table_schema` in `perspective.html`. @texodus Does this seem like a reasonable temporary fix to you?

I also took a look at the original code taken from perspective. There, this formatting is done on the basis of `table_schema`, and not the view's `schema`.